Sources/MapboxCoreNavigation/EHorizon/RoadGraphEdgeMetadata.swift

```diff
@@ -107,6 +110,29 @@ extension RoadGraph.Edge {
             self.drivingSide = drivingSide
             self.directionality = directionality
         }
+        
+        /**
+         Initializes a new edge `Metadata` object.
+         */
+        convenience init(heading: CLLocationDegrees,
+                         length: CLLocationDistance,
+                         roadClasses: RoadClasses,
+                         mapboxStreetsRoadClass: MapboxStreetsRoadClass,
+                         speedLimit: Measurement<UnitSpeed>?,
+                         speed: CLLocationSpeed,
+                         isBridge: Bool,
+                         names: [RoadName],
+                         laneCount: UInt?,
+                         altitude: CLLocationDistance?,
+                         curvature: UInt,
+                         countryCode: String?,
+                         regionCode: String?,
+                         drivingSide: DrivingSide,
+                         directionality: Directionality,
+                         isUrban: Bool) {
+            self.isUrban = isUrban
```

Comment:
[Builds](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/9124/workflows/5a63f89c-f728-4728-aaaf-ac3ff0fbd9b2/jobs/81967) are failing:

```
/Users/distiller/project/Sources/MapboxCoreNavigation/EHorizon/RoadGraphEdgeMetadata.swift:117:21: error: initializers in structs are not marked with 'convenience'
        convenience init(heading: CLLocationDegrees,
        ~~~~~~~~~~~~^
        
/Users/distiller/project/Sources/MapboxCoreNavigation/EHorizon/RoadGraphEdgeMetadata.swift:133:18: error: 'let' property 'isUrban' may not be initialized directly; use "self.init(...)" or "self = ..." instead
            self.isUrban = isUrban
                 ^
/Users/distiller/project/Sources/MapboxCoreNavigation/EHorizon/RoadGraphEdgeMetadata.swift:77:20: note: 'isUrban' declared here
        public let isUrban: Bool
```

I think the issue is that the convenience initializer is taking care of setting `isUrban`, but the required initializer is supposed to be the one that sets all the instance variables. Normally the convenience initializer would pass a hard-coded `isUrban` argument into the required initializer, the opposite of what we have here.

Reply:
From the backwards compatibility checker’s perspective, it might be a breaking change to relegate the original initializer’s signature to a convenience initializer, but I don’t think it would be a problem in practice: code that doesn’t specify the new parameter would end up calling the convenience initializer instead of the required initializer. In the past, there was an additional problem that subclasses would break b/c they suddenly wouldn’t override the right initializer anymore. But you can’t subclass a struct. The other consideration is if some code refers to the initializer symbolically, but that seems like a remote possibility for these two structs.